lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php

```diff
@@ -56,6 +56,11 @@ public function __construct(array $params, $username, $password, array $driverOp
 
         $socket = $params['unix_socket'] ?? ini_get('mysqli.default_socket');
         $dbname = $params['dbname'] ?? '';
+        $host   = $params['host'];
+
+        if (! empty($params['persistent'])) {
```

Comment:
I'd prefer `array_key_exists()` over more dangerous `! empty()`.

Reply:
In this case it's irrelevant. `array_key_exists()` over a mistyped key will have the same effect.

Reply:
I'm thinking of introducing a `ConnectionParameters` object which will be just a typed structure which will also contain username and password if they are specified. It could avoid optional `$username` and `$pasword` in driver/connection methods. Any suggestions/ideas?